src/store/uploads/uploads.provider.js

```diff
@@ -0,0 +1,64 @@
+import { storage } from '@/firebase'
+import { ref, uploadBytesResumable } from 'firebase/storage'
+
+const variants = {
+  sucess: 'success',
+  error: 'error',
+  loading: 'loading',
+}
+
+export const startUpload = (setUploads, files) => {
+  files.forEach((file, currentIndex) => {
+    // TODO: show error message if file type is not allowed
+
+    const songsRef = ref(storage, `songs/${file.name}`)
+    const task = uploadBytesResumable(songsRef, file)
+
+    const newUpload = {
+      task,
+      currentProgress: 0,
+      name: file.name,
+      variant: variants.loading,
+      icon: 'fas fa-spinner fa-spin',
+    }
+
+    setUploads((prev) => [newUpload, ...prev])
```

Comment:
A local state should not be handled / mixed with a provider, the provider must be almost like a black box, it does not matter how is the implementation inside of it.